test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/Implementation/ExportClient/OtlpExporterCompressionTests.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Buffers.Binary;
+using System.IO.Compression;
+using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ExportClient;
+using Xunit;
+
+#if !NET
+using System.Net.Http;
+#endif
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests.Implementation.ExportClient;
+
+public class OtlpExporterCompressionTests
+{
+    [Theory]
+    [InlineData(OtlpExportCompression.Gzip, "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA")]
+    [InlineData(OtlpExportCompression.None, "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA")]
+    [InlineData(OtlpExportCompression.Gzip, "")]
+    public void SendExportRequest_SendsCorrectContent_Http(OtlpExportCompression compression, string text)
+    {
+        var buffer = System.Text.Encoding.UTF8.GetBytes(text);
+
+        // Arrange
+        var options = new OtlpExporterOptions
+        {
+            Endpoint = new Uri("http://localhost:4317"),
+            Compression = compression,
+        };
+
+        using var testHttpHandler = new TestHttpMessageHandler();
+        using var httpClient = new HttpClient(testHttpHandler, false);
+        var exportClient = new OtlpHttpExportClient(options, httpClient, string.Empty);
+
+        var deadlineUtc = DateTime.UtcNow.AddMilliseconds(httpClient.Timeout.TotalMilliseconds);
+
+        // Act
+        var result = exportClient.SendExportRequest(buffer, buffer.Length, deadlineUtc);
+        var httpRequest = testHttpHandler.HttpRequestMessage;
+
+        // Assert
+        Assert.True(result.Success);
+        Assert.NotNull(httpRequest);
+        Assert.Equal(HttpMethod.Post, httpRequest.Method);
+        Assert.NotNull(httpRequest.Content);
+
+        if (compression == OtlpExportCompression.Gzip)
+        {
+            Assert.Contains(httpRequest.Content.Headers, h => h.Key == "Content-Encoding" && h.Value.First() == "gzip");
+
+            var content = testHttpHandler.HttpRequestContent;
+            Assert.NotNull(content);
+            var decompressedStream = Decompress(content);
+
+            Assert.Equal(buffer, decompressedStream.ToArray());
+        }
+        else
+        {
+            Assert.DoesNotContain(httpRequest.Content.Headers, h => h.Key == "Content-Encoding");
+            Assert.Equal(buffer, testHttpHandler.HttpRequestContent);
+        }
```

Comment:
Rather than conditional assertions in these tests, I think it would be better to split the tests into "not compressed" and "compressed" (even if the latter is only Gzip at the moment).

The shared logic between the two can just be a private helper method the tests can call.